File: Python/601-650/SetMismatch.py
class Solution:
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        s = set([i for i in range(1, len(nums) + 1)])
        ret = []
        for i in nums:
            if i in s:
                s.remove(i)
            else:
                ret.append(i)
        ret.append(s.pop())
        return ret

File: Java/601-650/SetMismatch.java
public class SetMismatch {
    public int[] findErrorNums(int[] nums) {
        long s = 0;
        int rep = 0;
        int n = nums.length;
        for (int i : nums) {
            int j = Math.abs(i) - 1;
            s += j + 1;
            if (nums[j] < 0) rep = j + 1;
            else nums[j] = -nums[j];
        }
        return new int[] { rep, (int)(n * (long)(n+1) / 2 - s + rep) };
    }
}
